aggregator/agg_functions.h

```diff
@@ -46,6 +46,46 @@
 #ifndef AGGREGATOR_AGG_FUNCTIONS_H
 #define AGGREGATOR_AGG_FUNCTIONS_H
 
+#include <climits>
+#include "output.h"
+
+/**
+ * Control function which checks signed data type T overflow when doing addition.
+ * @tparam T Template type variable, have to be signed (eg. int).
+ * @param a [in] Pointer to first operand.
+ * @param b [in] Pointer to second operand.
+ * @return Return 0 when no overflow occurs, different value (mostly 1) otherwise.
+ */
+template<typename T>
+int check_safe_signed_add(const void *a, const void *b) {
+   //T, T
+   int bits = sizeof(T) * CHAR_BIT;
+   uint64_t r = (uint64_t)(*((T*)a)) + (uint64_t)(*((T*)b));
+   return (((uint64_t)(*((T*)a)) ^ r) & ((uint64_t)(*((T*)b)) ^ r)) >> (bits - 1);
```

Comment:
I'd suggest to put some reference for these two lines...

Additionally, there should be some more spaces according to coding style.